R/analyzeVariablePlan.R

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright (C) 2018 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+##----------------------------------------------------------------
+##                    Analyze variable plan.                    --
+##----------------------------------------------------------------
+#' @param jaspResults {object} Object that will contain all results from the analysis and connect it to the output.
+#' @param dataset {object} (optional) tabular data, if available for the analysis.
+#' @param options {list} A named list of interface options selected by the user.
+##----------------------------------------------------------------
+AnalyzeVariablePlan <- function(jaspResults, dataset = NULL, options, ...) {
+  # Dependency variables
+  plan_vars <- c("sampleSize", "kValue")
+  
+  # Check if the container already exists. Create it if it doesn't.
+  if (is.null(jaspResults[["analyzeVarContainer"]]) || jaspResults[["analyzeVarContainer"]]$getError()) {
+    analyzeVarContainer <- createJaspContainer(title = "Analyze Variable Plan")
+    analyzeVarContainer$dependOn(plan_vars) # Common dependencies
+    jaspResults[["analyzeVarContainer"]] <- analyzeVarContainer
+  } else {
+    analyzeVarContainer <- jaspResults[["analyzeVarContainer"]]
+  }
+  # Plan variables
+  N <- options$lotSize
+  n <- options$sampleSize
+  k <- options$kValue
+  sd <- "unknown"
+  if (options$sd) {
+    sd <- "known"
+  }
+  
+  # Initialize the plan table
+  plan_table <- createJaspTable(title = gettextf("Variable Sampling Plan (Standard deviation assumed to be <b>%s</b>)", sd))
+  plan_table$transpose <- TRUE
+  plan_table$transposeWithOvertitle <- FALSE
+  plan_table$dependOn(c(plan_vars, "sd"))
+  plan_table$addColumnInfo(name = "col_0", title = "", type = "string") # Dummy row
+  plan_table$addColumnInfo(name = "col_1", title = "Sample size", type = "integer")
+  plan_table$addColumnInfo(name = "col_2", title = "Critical Distance (k)", type = "number")
+  plan_table$addRows(list("col_1" = n, "col_2" = k))
+  plan_table$showSpecifiedColumnsOnly <- TRUE
+  plan_table$position <- 1
+  analyzeVarContainer[["plan_table"]] <- plan_table
+
+  # Error check for n (sample size)
+  if (!options$sd && (n <= 1)) {
+    analyzeVarContainer$setError(sprintf("If historical standard deviation is unknown, sample size has to be > 1."))
```

Comment:
Yes, it can be, but I still think it's easier to maintain it this way, and also easier to understand.
I've used QML constraints where they're simple enough, but extended checks in QML code (making one variable's value depend on another) can become a bit clunky!